tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorTests.cs

```diff
@@ -147,6 +144,48 @@ public void Any_with_a_nested_Any()
                 "{G: {$elemMatch: {D: \"Don't\", \"S\": {$elemMatch: {E: null, D: \"Delilah\" }}}}}");
         }
 
+        [Fact]
+        public void Any_with_a_not_and_a_predicate_with_not_contains()
+        {
+            var x = new[] { 1, 2 };
+
+            Assert(
+                c => !c.M.Any(a => !x.Contains(a)),
+                "{ M : { '$not' : { '$elemMatch' : { '$nin' : [1, 2] } } } }");
+        }
+
+        [Fact]
+        public void Any_with_a_not_and_a_predicate_with_contains()
+        {
+            var x = new[] { 1, 2 };
+
+            Assert(
+                c => !c.M.Any(a => x.Contains(a)),
+                "{ M : { '$not' : { '$elemMatch' : { '$in' : [1, 2] } } } }");
+        }
+
+        [Fact]
+        public void Any_with_a_predicate_with_contains()
+        {
+            var x = new[] { 1, 2 };
+
+            Assert(
+                c => c.M.Any(a => x.Contains(a)),
+                "{ M : { '$elemMatch' : { '$in' : [1, 2] } } }"
```

Comment:
Why do we need $elemMatch here at all?

Reply:
Just thought that `{ M : {'$in' : [1, 2] } }` would do the same thing. Perhaps it doesn't.

Reply:
The queries without **negation** : `{ M : {'$in' : [1, 2] } }` and `{ M : { '$elemMatch' : { '$in' : [1, 2] } } }` wil have the same result:

`db.test.find({ a : { '$elemMatch' : { '$in' : [1, 2] } } })
{ "_id" : 2, "a" : [ 1 ] }
{ "_id" : 3, "a" : [ 2 ] }
{ "_id" : 6, "a" : [ 1, 2 ] }
{ "_id" : 7, "a" : [ 1, 3 ] }
{ "_id" : 8, "a" : [ 1, 4 ] }
{ "_id" : 9, "a" : [ 2, 3 ] }
{ "_id" : 11, "a" : [ 1, 2, 3 ] }
{ "_id" : 12, "a" : [ 1, 2, 4 ] }
{ "_id" : 13, "a" : [ 1, 3, 4 ] }
{ "_id" : 14, "a" : [ 1, 2, 3, 4 ] }`
and
`db.test.find({ a : { '$in' : [1, 2] } } )
{ "_id" : 2, "a" : [ 1 ] }
{ "_id" : 3, "a" : [ 2 ] }
{ "_id" : 6, "a" : [ 1, 2 ] }
{ "_id" : 7, "a" : [ 1, 3 ] }
{ "_id" : 8, "a" : [ 1, 4 ] }
{ "_id" : 9, "a" : [ 2, 3 ] }
{ "_id" : 11, "a" : [ 1, 2, 3 ] }
{ "_id" : 12, "a" : [ 1, 2, 4 ] }
{ "_id" : 13, "a" : [ 1, 3, 4 ] }
{ "_id" : 14, "a" : [ 1, 2, 3, 4 ] }`

Reply:
The queries with negation  has different results:
``` db.test.find({ a : { '$elemMatch' : { '$nin' : [1, 2] } } })
{ "_id" : 4, "a" : [ 3 ] }
{ "_id" : 5, "a" : [ 4 ] }
{ "_id" : 7, "a" : [ 1, 3 ] }
{ "_id" : 8, "a" : [ 1, 4 ] }
{ "_id" : 9, "a" : [ 2, 3 ] }
{ "_id" : 10, "a" : [ 3, 4 ] }
{ "_id" : 11, "a" : [ 1, 2, 3 ] }
{ "_id" : 12, "a" : [ 1, 2, 4 ] }
{ "_id" : 13, "a" : [ 1, 3, 4 ] }
{ "_id" : 14, "a" : [ 1, 2, 3, 4 ] }```
and
```db.test.find({ a : {'$nin' : [1, 2] } })
{ "_id" : 111, "a" : [ ] }
{ "_id" : 4, "a" : [ 3 ] }
{ "_id" : 5, "a" : [ 4 ] }
{ "_id" : 10, "a" : [ 3, 4 ] }```